src/ERC20Permit.sol

```diff
@@ -4,12 +4,13 @@ pragma solidity 0.8.23;
 
 import { IERC20 } from "./interfaces/IERC20.sol";
 import { IERC20Permit } from "./interfaces/IERC20Permit.sol";
+import { ERC712 } from "./libs/ERC712.sol";
 
-import { StatefulERC712 } from "./StatefulERC712.sol";
+import { ERC3009 } from "./ERC3009.sol";
 
 /// @title Permit Extension for ERC20 Signed Approvals via EIP-712 with EIP-2612 and EIP-1271 compatibility.
 /// @dev   An abstract implementation to satisfy EIP-2612: https://eips.ethereum.org/EIPS/eip-2612
-abstract contract ERC20Permit is IERC20Permit, StatefulERC712 {
+abstract contract ERC20Permit is IERC20Permit, ERC3009 {
```

Comment:
If `ERC20Permit is ERC3009`, then needs to be `IERC20Permit is IERC3009`.

Reply:
Fixed in this commit: https://github.com/MZero-Labs/common/pull/7/commits/287b68343734e04236e9d631ee9a4d0e8db50a21